Build failure: python311Packages.behave on Darwin

### Steps To Reproduce

1. build python311Packages.behave on Darwin
2. Observe system failure 2 during testing:
```sh
tests/api/test_async_step.py ....sss...                                  [  0%]
tests/functional/test_active_tags.py ................................... [  3%]
..............                                                           [  4%]
tests/functional/test_tag_expression.py ................................ [  6%]
............................                                             [  8%]
tests/functional/test_tag_inheritance.py .........................       [ 10%]
tests/issues/test_issue0336.py ..                                        [ 10%]
tests/issues/test_issue0449.py ...                                       [ 10%]
tests/issues/test_issue0453.py ...                                       [ 10%]
tests/issues/test_issue0458.py ....                                      [ 11%]
tests/issues/test_issue0495.py ...                                       [ 11%]
tests/issues/test_issue0619.py .                                         [ 11%]
tests/issues/test_issue0725.py .                                         [ 11%]
tests/issues/test_issue0767.py .                                         [ 11%]
tests/issues/test_issue1047.py ..                                        [ 11%]
tests/issues/test_issue1054.py ..                                        [ 11%]
tests/issues/test_issue1061.py ...                                       [ 11%]
tests/unit/reporter/test_summary.py .........                            [ 12%]
tests/unit/tag_expression/test_basics.py .................               [ 13%]
tests/unit/tag_expression/test_model.py ................................ [ 16%]
...............................................                          [ 19%]
tests/unit/tag_expression/test_model_ext.py ................             [ 20%]
tests/unit/tag_expression/test_parser.py ............................... [ 22%]
...........................................................              [ 26%]
tests/unit/tag_expression/test_tag_expression_v1_part1.py .............. [ 27%]
........................................................................ [ 32%]
........................................................................ [ 37%]
...                                                                      [ 38%]
tests/unit/tag_expression/test_tag_expression_v1_part2.py .............. [ 39%]
....                                                                     [ 39%]
tests/unit/test_ansi_escapes.py ........................................ [ 42%]
...........................................................              [ 46%]
tests/unit/test_behave4cmd_command_shell_proc.py ........xx....          [ 47%]
tests/unit/test_capture.py ..........................                    [ 49%]
tests/unit/test_configuration.py .................FFFFFF.......          [ 51%]
tests/unit/test_context_cleanups.py ................                     [ 52%]
tests/unit/test_explore_generator.py ....                                [ 52%]
tests/unit/test_fixture.py ....................................          [ 55%]
tests/unit/test_formatter.py .................................           [ 57%]
tests/unit/test_formatter_progress.py .................................  [ 59%]
tests/unit/test_formatter_rerun.py ...........................           [ 61%]
tests/unit/test_formatter_tags.py ...................................... [ 64%]
.                                                                        [ 64%]
tests/unit/test_importer.py ...........                                  [ 65%]
tests/unit/test_log_capture.py .                                         [ 65%]
tests/unit/test_matchers.py .....................................        [ 68%]
tests/unit/test_model.py ............................................... [ 71%]
......                                                                   [ 71%]
tests/unit/test_model2.py .........................                      [ 73%]
tests/unit/test_model_core.py .....................................      [ 76%]
tests/unit/test_parser.py .............................................. [ 79%]
.......................                                                  [ 81%]
tests/unit/test_parser_gherkin_v6.py .......................             [ 82%]
tests/unit/test_runner.py .............................................. [ 85%]
.............                                                            [ 86%]
tests/unit/test_runner_plugin.py ..............                          [ 87%]
tests/unit/test_runner_util.py ............                              [ 88%]
tests/unit/test_step_registry.py .....                                   [ 88%]
tests/unit/test_tag_matcher.py ......................................... [ 91%]
............................                                             [ 93%]
tests/unit/test_textutil.py ...............sss..ss..ss                   [ 95%]
tests/unit/test_userdata.py ............................................ [ 98%]
.................                                                        [100%]

=================================== FAILURES ===================================
_______________ TestConfigFile.test_tag_expression_protocol[any] _______________

self = <tests.unit.test_configuration.TestConfigFile object at 0x106a4e710>
value = 'any'
tmp_path = PosixPath('/private/tmp/nix-build-python3.11-behave-1.2.7.dev5.drv-0/pytest-of-_nixbld1/pytest-0/test_tag_expression_protocol_a0')

    @pytest.mark.parametrize("value", TagExpressionProtocol.choices())
    def test_tag_expression_protocol(self, value, tmp_path):
>       self.check_tag_expression_protocol_with_valid_value(value, tmp_path)

tests/unit/test_configuration.py:313:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/unit/test_configuration.py:303: in check_tag_expression_protocol_with_valid_value
    config = Configuration()
behave/configuration.py:789: in __init__
    args = parser.parse_args(command_args)
/nix/store/k07bkf79yb122qa1hkh3nis7q3zim3b6-python3-3.11.10/lib/python3.11/argparse.py:1877: in parse_args
    self.error(msg % ' '.join(argv))
/nix/store/k07bkf79yb122qa1hkh3nis7q3zim3b6-python3-3.11.10/lib/python3.11/argparse.py:2640: in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ArgumentParser(prog='__main__.py', usage='%(prog)s [options] [DIRECTORY|FILE|FILE:LINE|AT_FILE]*', description='Run a ...atures.txt\n', formatter_class=<class 'argparse.RawDescriptionHelpFormatter'>, conflict_handler='error', add_help=True)
status = 2, message = '__main__.py: error: unrecognized arguments: -k\n'

    def exit(self, status=0, message=None):
        if message:
            self._print_message(message, _sys.stderr)
>       _sys.exit(status)
E       SystemExit: 2

/nix/store/k07bkf79yb122qa1hkh3nis7q3zim3b6-python3-3.11.10/lib/python3.11/argparse.py:2627: SystemExit
----------------------------- Captured stderr call -----------------------------
usage: __main__.py [options] [DIRECTORY|FILE|FILE:LINE|AT_FILE]*
__main__.py: error: unrecognized arguments: -k
```

### Additional context

The underlying problem seems to be the `-k` argument being passed causes a hard exit. 1.2.7dev6 was released six weeks ago, I wonder if it would help?

### Notify maintainers

@alunduil 
@maxxk 

### Metadata

```sh
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```


---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/behave/default.nix
+++ b/pkgs/development/python-modules/behave/default.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   buildPythonPackage,
   python,
   pythonOlder,
@@ -30,6 +31,14 @@ buildPythonPackage rec {
     rev = "v${version}";
     hash = "sha256-G1o0a57MRczwjGLl/tEYC+yx3nxpk6+E58RvR9kVJpA=";
   };
+
+  patches = [
+    # fix tests: https://github.com/behave/behave/pull/1214
+    (fetchpatch2 {
+      url = "https://github.com/behave/behave/pull/1214/commits/98b63a2524eff50ce1dc7360a46462a6f673c5ea.patch?full_index=1";
+      hash = "sha256-MwODEm6vhg/H8ksp5XBBP5Uhu2dhB5B1T6Owkxpy3v0=";
+    })
+  ];
 
   build-system = [ setuptools ];
 

